### PR TITLE
Optimize memory usage in `DeepStateModel` and `DeepARNativeModel` by eliminating copying during making samples

### LINKED ISSUE #496: Optimize memory usage in `DeepStateModel` and `DeepARNativeModel ` by eliminating copying during making samples
### 🚀 Task description

Now `DeepStateModel` and `DeepARNativeModel ` uses enormous amount memory when applied to big datasets. It occurs when samples are generating. It happens because samples consist of arrays copies instead of views.

For `DeepStateModel` samples are immediately converted to `torch.Tensor`, see https://github.com/etna-team/etna/blob/950f676c24e3f3b67917991f63200fa3482b9d84/etna/models/nn/deepstate/deepstate.py#L232

For `DeepARModel` "real" parts of sample are copies, see https://github.com/etna-team/etna/blob/950f676c24e3f3b67917991f63200fa3482b9d84/etna/models/nn/deepar_native/deepar.py#L258

### Plan

For `DeepStateModel` convert tensors to arrays

For `DeepARModel` delete copies

### Test cases

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by egoriyaa**
## Before submitting (must do checklist)

- [ ] Did you read the [contribution guide](https://github.com/etna-team/etna/blob/master/CONTRIBUTING.md)?
- [ ] Did you update the docs? We use Numpy format for all the methods and classes.
- [ ] Did you write any new necessary tests?
- [ ] Did you update the [CHANGELOG](https://github.com/etna-team/etna/blob/master/CHANGELOG.md)?

## Closing issues
closes #496

### CODE REVIEW

File: etna/models/nn/deepstate/deepstate.py [hunk 1]
```diff
@@ -117,9 +118,9 @@ def step(self, batch: DeepStateBatch, *args, **kwargs):  # type: ignore
         :
             loss, true_target, prediction_target
         """
-        encoder_real = batch["encoder_real"]  # (batch_size, seq_length, input_size)
+        encoder_real = batch["encoder_real"].float()  # (batch_size, seq_length, input_size)
```
[the remark this answers is not in the document]
Comment (egoriyaa) [reply etna/models/nn/deepstate/deepstate.py:121]: We do it everywhere. As I understand, numpy and torch have different "understanding" of  float types. If we don't do this, exception is raised when passing to any torch layer.

File: etna/models/nn/deepstate/deepstate.py [hunk 2]
```diff
@@ -213,26 +214,23 @@ def forward(self, x: DeepStateBatch, *args, **kwargs):  # type: ignore
...
 
         # Categories that were not seen during `fit` will be filled with new category
         for feature in self.embedding_sizes:
             df[feature] = df[feature].astype(float).fillna(self.embedding_sizes[feature][0])
 
         # Columns in `values_categorical` are in the same order as in `embedding_sizes`
-        values_categorical = torch.from_numpy(df[self.embedding_sizes.keys()].values.T)
+        values_categorical = df[self.embedding_sizes.keys()].values.T
 
-        values_datetime = torch.from_numpy(self.ssm.generate_datetime_index(df["timestamp"]))
```
[the remark this answers is not in the document]
Comment (egoriyaa) [reply etna/models/nn/deepstate/deepstate.py:225]: Also we have such signature for all our model. I understand it as the signature of generated batch, not sample

File: CHANGELOG.md [hunk 3]
```diff
@@ -40,7 +40,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Fix working with `embedding_sizes` in `202-NN_examples` notebook ([#489](https://github.com/etna-team/etna/pull/489))
 - Disallow dropping target in `TSDataset.drop_features` ([#491](https://github.com/etna-team/etna/pull/491))
 - Optimize memory usage in `TFTNativeModel` by eliminating copying during making samples ([#494](https://github.com/etna-team/etna/pull/494))
-- 
+- Optimize memory usage in `DeepStateModel` and `DeepARModel` by eliminating copying during making samples ([#499](https://github.com/etna-team/etna/pull/499))
```
[the remark this answers is not in the document]
Comment (d-a-bunin) [reply CHANGELOG.md]: Let's also update name of issue and name of the PR.

File: etna/models/nn/deepstate/deepstate.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (d-a-bunin) [reply etna/models/nn/deepstate/deepstate.py:121]: How did it work before this?

File: etna/models/nn/deepstate/deepstate.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (d-a-bunin) [reply etna/models/nn/deepstate/deepstate.py:225]: Ok, understood.

File: etna/models/nn/deepstate/deepstate.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (d-a-bunin) [reply etna/models/nn/deepstate/deepstate.py:121]: Which type is it before convertation?

**Review by d-a-bunin — APPROVED** [review]

**Pull request merged**